dashboard/src/modules/components/DatePickerComponent/index.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState } from "react";
+import {
+  InputGroup,
+  InputGroupText,
+  DatePicker,
+  isValidDate,
+  yyyyMMddFormat,
+  Button,
+} from "@patternfly/react-core";
+import moment from "moment";
+
+function DatePickerWidget({
+  dataArray,
+  setPublicData,
+  controllerName,
+  setDateRange,
+}) {
+  const [fromDate, setFromDate] = useState(
+    moment(new Date(1990, 10, 4)).format("YYYY/MM/DD")
+  );
+  const [toDate, setToDate] = useState(
+    moment(new Date(2040, 10, 4)).format("YYYY/MM/DD")
+  );
+  const toValidator = (date) =>
+    isValidDate(fromDate) && date >= fromDate
+      ? ""
+      : "To date must be less than from date";
+  let modifiedArray = [];
+  const onFromChange = (_str, date) => {
+    setFromDate(new Date(date));
+    moment(date).format("DD/MM/YYYY");
+    if (isValidDate(date)) {
+      date.setDate(date.getDate() + 1);
+      setToDate(yyyyMMddFormat(date));
+    } else {
+      setToDate("");
+    }
+  };
+
+  const filterByDate = () => {
+    modifiedArray = dataArray.filter((data) => {
+      let formattedData = moment(data.metadata["dataset.created"]).format(
+        "YYYY/MM/DD"
+      );
+      return (
+        Date.parse(formattedData) >= Date.parse(fromDate) &&
+        Date.parse(formattedData) <= Date.parse(toDate) &&
```

Comment:
Can't the basic date objects be compared without formatting them first?  This seems like a pretty normal thing to do, and rendering them into a formatted string so that you can compare the strings seems like wasted effort.

Reply:
We also need to think about when we're going to re-fetch data from the server. I think this is oriented towards filtering the data we already have (for example in a Redux store) to control what's displayed. If we do this for long, it'll be out of date. A cache is good, but a Redux store will persist across page reloads.

You *could* do a new query with the `after` and `before` query parameters. We need to decide when we want to do that vs continuing to filter a cached list.

Reply:
@dbutenhof, that sounds good, but in terms of this PR, I think we should be recommending that @shubham-html-css-js take the value returned by the server and convert it to a Javascript `Date` object and then compare that object to the `fromDate` and `toDate` objects directly, without formatting them as strings first.

Reply:
Hey, standard server JSON date format was your tangent, not mine. 😆 

I agree, the dashboard should pull everything into standard Javascript `Date` objects and compare those.